tock/hours/tests/test_views.py

```diff
@@ -599,6 +599,35 @@ def test_report_list_authenticated(self):
         )
         self.assertEqual(response.status_code, 200)
 
+    def test_auto_create_reporting_period(self):
```

Comment:
i would also add a test or two that explicitly test your disallowed_date function to make sure the ranges return are what's expected when supplied a few diff dates